### ISSUE TITLE
Import/export all settings

### ISSUE THREAD

**Issue by Bionus** [Bionus/imgbrd-grabber#583]
Sometimes, when wanting to do a fresh install (issue #582 for example), or when moving to another computer, we may want to import/export our settings.

*Note: first step for doing suggestion #531*

**Comment by Bionus** [comment]
Things to save:
* Settings (`settings.ini`)
* Favorites (`favorites.txt`)
* Favorites' thumbnails (`thumbs/*.png`)
* Ignored tags (`ignore.txt`)
* View it later tags (`viewitlater.txt`)
* Custom words (`wordsc.txt`)
* Site list (`sites/*/sites.txt`)
* Site settings (`sites/*/*/settings.ini`)

**Comment by barryanders** [comment]
I like this simply as a means of back up so you can always restore your settings no matter what happens. I've already been backing up my user files manually so nice plan.

Also on a side note, thanks for working on the single image navigation. I saw some commits in the development. I'll have to check that out today some time.

**Comment by Bionus** [comment]
> I like this simply as a means of back up so you can always restore your settings no matter what happens. I've already been backing up my user files manually so nice plan.

Yes, that's the main objective. Note that it won't save some computer-related stuff, such as the md5 list, the filename history, etc. These are not really required in my opinion.

> Also on a side note, thanks for working on the single image navigation. I saw some commits in the development. I'll have to check that out today some time.

It will be included in the 5.0.0, as well as a scrollable results area. I actually planned to release it this week-end, but it didn't manage to finish wrapping everything up until a few minutes ago, so I guess the official version be for next week. I'd rather not push a release a Sunday at midnight.

I uploaded an alpha though, but since you're on Mac I don't think that will be of much help. :smile:

**Comment by barryanders** [comment]
lol, when you get to a solid 5 that you feel is stable enough to not be beta, I'll build one again for other users.

**Comment by bahari15** [comment]
> Things to save:
> 
> * Settings (`settings.ini`)
> * Favorites (`favorites.txt`)
> * Favorites' thumbnails (`thumbs/*.png`)
> * Ignored tags (`ignore.txt`)
> * View it later tags (`viewitlater.txt`)
> * Custom words (`wordsc.txt`)
> * Site list (`sites/*/sites.txt`)
> * Site settings (`sites/*/*/settings.ini`)

Hi, I still cannot find these files that you referenced in my installation directory, is there something wrong or changed?

**Comment by dumbuz** [comment]
Seems to exist.
Options > Backup > Generate a backup
To restore, extract the zip file and open the restore.igl file with grabber closed.
